internal/encoding/parser.go

```diff
@@ -0,0 +1,90 @@
+package encoding
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/internal/encoding/graphite"
+)
+
+type Parser struct {
+	graphiteParser *graphite.Parser
+}
+
+func NewParser(parser *graphite.Parser) *Parser {
+	return &Parser{graphiteParser: parser}
+}
+
+func (p *Parser) Parse(dataFormat string, out []byte, acc telegraf.Accumulator) error {
+	var err error
+	var metrics []telegraf.Metric
+	var metric telegraf.Metric
+
+	switch dataFormat {
+	case "", "json":
```

Comment:
Make this case only `case "json":`, if plugins want to convert "" -> "json" they can make that assumption themselves.